src/cart/services/cart.service.ts

```diff
@@ -34,22 +47,23 @@ export class CartService {
     return this.createByUserId(userId);
   }
 
-  updateByUserId(userId: string, { items }: Cart): Cart {
-    const { id, ...rest } = this.findOrCreateByUserId(userId);
+  async updateByUserId(
+    userId: string,
+    updatedCart: Partial<Cart>,
+  ): Promise<Cart> {
+    let cart = await this.findByUserId(userId);
 
-    const updatedCart = {
-      id,
-      ...rest,
-      items: [ ...items ],
-    }
+    cart = {
+      ...cart,
+      ...updatedCart,
+    };
 
-    this.userCarts[ userId ] = { ...updatedCart };
+    await this.cartRepository.save(cart);
```

Comment:
Have you tested this method? Did you have a problem with duplicate orphaned rows in the database after executing the save method?